test-data/unit/pythoneval.test

```diff
@@ -27,6 +27,18 @@ print(list(reversed(A())))
 \['c', 'b', 'a']
 \['f', 'o', 'o']
 
+[case testOldIterableAndIterableProtocolAreIncompatible]
```

Comment:
This is something I want to discuss.
I can make `Iterable` and `__getitem__` behave like subtypes. But this will be a lie!

Because `Iterable` suppose to have `__iter__` method, which does not really exist on classes with only `__getitem__`.

So, any direct invokations of `iterable.__iter__` will fail.

What should we do here?
1. Make life easier and accept `__getitem__` everywhere where `Iterable` is expected
2. Keep it as-is
3. Find some middle-ground. For example, accept `__getitem__` everywhere where `Iterable` is expected, but be very strict about explicit `.__iter__` method access.

Reply:
However, `2.` does not make much sence. It won't solve problems with `enumerate`, `list`, and custom functions, where iteration is used.